ibis/expr/types/relations.py

```diff
@@ -144,6 +144,113 @@ def __contains__(self, name: str) -> bool:
         """
         return name in self.schema()
 
+    def cast(
+        self,
+        types: Iterable[dt.DataType | str]
+        | sch.Schema
+        | Mapping[str, dt.DataType | str],
+    ) -> Table:
+        """Cast the columns of a table.
+
+        !!! note "If you need to cast columns to a single type, use [selectors](https://ibis-project.org/blog/selectors/)."
+
+        Parameters
+        ----------
+        types
+            List of types, mapping or schema to use for casting
+
+        Returns
+        -------
+        Table
+            Casted table
+
+        Examples
+        --------
+        >>> import ibis
+        >>> import ibis.expr.selectors as s
+        >>> ibis.options.interactive = True
+        >>> t = ibis.examples.penguins.fetch()
+        >>> t.schema()
+        ibis.Schema {
+          species            string
+          island             string
+          bill_length_mm     float64
+          bill_depth_mm      float64
+          flipper_length_mm  int64
+          body_mass_g        int64
+          sex                string
+          year               int64
+        }
+        >>> cols = ["body_mass_g", "bill_length_mm"]
+        >>> t[cols].head()
+        ┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┓
+        ┃ body_mass_g ┃ bill_length_mm ┃
+        ┡━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━┩
+        │ int64       │ float64        │
+        ├─────────────┼────────────────┤
+        │        3750 │           39.1 │
+        │        3800 │           39.5 │
+        │        3250 │           40.3 │
+        │           ∅ │            nan │
+        │        3450 │           36.7 │
+        └─────────────┴────────────────┘
+        >>> expr = t.cast({"body_mass_g": "float64", "bill_length_mm": "int"})
```

Comment:
IIUC passing a schema (or schema-like object) doesn't have to include every column in the input table - columns that exist but aren't present in the specified schema are passed through unmodified. If this understanding is correct, it'd be nice to mention that somewhere in this docstring.